sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -462,16 +495,20 @@ export class BatchingReceiver extends MessageReceiver {
           onClose: onReceiveClose,
           onSessionClose: onSessionClose
         });
-        this._init(rcvrOptions)
+        this._init(rcvrOptions, abortSignal)
           .then(() => {
             if (!this._receiver) {
               // there's a really small window here where the receiver can be closed
               // if that happens we'll just resolve to an empty array of messages.
               return resolve([]);
             }
 
-            this._receiver.on(ReceiverEvents.receiverDrained, onReceiveDrain);
-            addCreditAndSetTimer();
+            // TODO: long-term we probably need to split the code in this promise. This check
+            // is just a band-aid for now.
+            if (!abortSignal?.aborted) {
```

Comment:
Should we close the receiver, or at least reject with the abortError if we get to this point and the receiver does exist?

Reply:
The receiver is actually not invalid so I think we're okay. The reason I avoid doing the above is because it basically initiates the "batch" receive on the receiver but we're going to tear everything down when we abort, so the receiver would be in an inconsistent state.

Reply:
But aren't we currently relying on `addCreditAndSetTime()` to resolve the outer promise? I'm worried that if we do have a valid receiver, and an aborted abortSignal, we'll never call execute the code path that calls resolve or reject, so it will just hang.

Reply:
You know...I'm just going to do some rearranging. I can do it without being too destructive and now that you know I'm doing it it'll be easy to review. :)

This will also fix your sessionError/cleanupBeforeReject comment as well.

Reply:
Rearranged, just for you! :)